### DISCUSSION TITLE
extract cascade temp

### DISCUSSION THREAD

**Discussion by Moods801** [sindrebroch/ha-flexit#23] [Ideas]
Great job on this integration, thank you.  
When first trying to use this integration it failed and the log showed me that I recieved response on 17 sensors:
DEBUG (MainThread) [custom_components.flexit] FlexitSensorsResponse. plant=xxxx. data={'totalCount': 17, 'values':  

Then I realized that I earlier changed the configuration on my Nordic S3 to regulate on extract cascade temp instead of the default config regulation on supply temp. 
This can be changed in the Flexit GO app when logging in as installer. The integration works after I changed this to the default.

I will leave the config as default for now and continue to use the integration, but if the integration can be updated to support this feature it would be great.

------------------------------------------------------------------------------------------

Logger: custom_components.flexit
Source: custom_components/flexit/models.py:99
Integration: Flexit (documentation, issues)
First occurred: 11:13:08 PM (6 occurrences)
Last logged: 11:15:46 PM

Unexpected error fetching flexit data: '(plant);1!0020007CA000055'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 187, in _async_refresh
    self.data = await self._async_update_data()
  File "/config/custom_components/flexit/coordinator.py", line 56, in _async_update_data
    return await self.api.sensor_data()
  File "/config/custom_components/flexit/api.py", line 145, in sensor_data
    return FlexitSensorsResponse.from_dict(
  File "/config/custom_components/flexit/models.py", line 202, in from_dict
    home_air_temperature=util._float_sensor(HOME_AIR_TEMPERATURE_PATH),
  File "/config/custom_components/flexit/models.py", line 107, in _float_sensor
    return round(float(self._str_sensor(path)), 2)
  File "/config/custom_components/flexit/models.py", line 99, in _str_sensor
    return self.data[VALUES][f"{self.plant}{path}"][VALUE][VALUE]
KeyError: '(plant);1!0020007CA000055'

**Comment by sindrebroch** [comment]
Interesting, didn't know this was a thing. How do you log in as an installer? Also curious of what difference this makes? 

I might add it to the roadmap if it doesn't seem too complicated, it might just be a different endpoint.

**Comment by Moods801** [comment]
You need to connect to the flexit device locally.
Page 4 of this PDF explains how.
https://www.flexit.no/globalassets/catalog/documents/26045_man_116628.pdf
It opens up alot of additional settings to adjust or values to display

**Comment by uphillbattle** [comment]
Og du fikk servicekoden uten å være installatør?

**Comment by kazzkizz** [comment]
> 
> 
> Og du fikk servicekoden uten å være installatør?

Logg inn med bruker 1000 mener jeg. Da er man installatør. Flexit oppgir dette i manualen.

**Comment by uphillbattle** [comment]
Ja, jaggu, så enkelt er det. Det var jo bare å prøve :) Takk for tipset!

**Comment by sindrebroch** [comment]
@Moods801 What setting did you change exactly, and where do I find it? Tried digging around a bit, but no luck. No promises on adding support, but want to check out how difficult it would be.

**Comment by sindrebroch** [comment]
Turning this into a Discussion as it is more of a feature request than a bug, and more information is needed to continue working on this.
